src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -357,5 +361,15 @@ public static int sortAndDedup(final BytesRefArray bytes, final int[] indices) {
 
     }
 
+    /**
+     * Combines multiple iterators into a single iterator.
+     */
+    public static <T> Iterator<T> concat(Iterator<? extends T>... iterators) {
```

Comment:
Because you don't want them to use the shaded Guava because its an implementation detail?